battenberg/core.py

```diff
@@ -77,27 +77,30 @@ def __exit__(self, type, value, traceback):
 
 class Battenberg:
 
-    def __init__(self, repo: Repository, context_file: str = '.cookiecutter.json'):
+    def __init__(self, repo: Repository):
         self.repo = repo
-        self.context_file = context_file
 
     def is_installed(self) -> bool:
         return TEMPLATE_BRANCH in self.repo.listall_branches()
 
-    def get_template(self):
-        context = self.get_context()
-        return context['_template']
-
-    def get_context(self) -> Dict[str, Any]:
-        with open(os.path.join(self.repo.workdir, self.context_file)) as f:
+    def get_context(self, context_file: str) -> Dict[str, Any]:
+        with open(os.path.join(self.repo.workdir, context_file)) as f:
             return json.load(f)
 
-    def merge_template_branch(self, message: str):
-        # Lookup template branch
+    def merge_template_branch(self, message: str, merge_target: str = None):
         branch = self.repo.lookup_branch(TEMPLATE_BRANCH)
 
-        # Analyze merge between template branch and HEAD
-        analysis, _ = self.repo.merge_analysis(branch.target)
+        merge_target_ref = 'HEAD'
+        if merge_target is not None:
+            # If we have a merge target, ensure we have that branch and have switched to it
+            # before continuing with merging.
+            merge_target_ref = f'refs/heads/{merge_target}'
+            if merge_target not in self.repo.branches:
+                self.repo.branches.local.create(merge_target, self.repo.get(self.repo.head.target))
+            self.repo.checkout(merge_target_ref)
+
+        # Analyze merge between template branch and our merge target.
```

Comment:
Nit: Is this comment adding information?

Reply:
Removed it